code/modules/client/client_procs.dm

```diff
@@ -574,7 +574,7 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 				"Forever alone :("\
 			)
 
-			send2adminchat("Server", "[cheesy_message] (No admins online)")
+			INVOKE_ASYNC(GLOBAL_PROC, GLOBAL_PROC_REF(send2adminchat), "Server", "[cheesy_message] (No admins online)")
```

Comment:
```suggestion
			ASYNC
				send2adminchat("Server", "[cheesy_message] (No admins online)")
```

Reply:
@LemonInTheDark didnt you not want this. you said something on the pr for it. i dont remember

Reply:
you would use it if you wanted to extract the return value from the proc if it didn't sleep.
I'm not sure why you'd do it here. There is a slight overhead to the invoke but it's minor, and this is damn depressing to read.

Reply:
you cannot get the return value from any code inside a spawn block, even if it doesn't actually block

Reply:
to be fair lemon the INVOKE_ASYNC is not more readable lol

Reply:
IMO the syntax should just be
```
INVOKE_ASYNC(proc())
```
the callback stuff is legacy and could well be converted.

> you cannot get the return value from any code inside a spawn block, even if it doesn't actually block

I hate this codebase

Reply:
Just merge the PR guys 😭

Reply:
> IMO the syntax should just be
> 
> ```
> INVOKE_ASYNC(proc(args))
> ```
> 
> the callback stuff is legacy and could well be converted.
> 
> > you cannot get the return value from any code inside a spawn block, even if it doesn't actually block
> 
> I hate this codebase

if i had the patient of a saint i wouldve done that, since i agree thats the superior syntax. throws out the need for PROC_REF in this context, and only requires it in callbacks